src/mesh/mesh_adaptation.h

```diff
@@ -51,6 +51,9 @@ class MeshAdaptation
     /// Performs fixed fraction refinement based on refinement and coarsening fractions.
     void fixed_fraction_isotropic_refinement_and_coarsening();
     
+    /// Decide whether to perform h or p refinement based on a smoothness indicator.
```

Comment:
It would be better to add a note here about using overintegration if hp-adaptation has issues with convergence.

Reply:
Are you sure this is the right place to leave the comment about overintegration?

Reply:
Hmm, ideally we would output the message when mesh_adaptation_type is set to hp_adaptation and the residual does not converge. Maybe you could do that in post processing of the flow solver.

Reply:
Maybe this PR should go through before #164 so that @MayaTatarelli doesn't have a bunch of merge conflicts

Reply:
Agreed, this PR should be merged before #164.